resolve.go

```diff
@@ -39,7 +41,13 @@ func resolveTransport(req *dns.Msg, transport, resolver string) (resp *dns.Msg,
 		resolver = resolver + ":53"
 	}
 
-	c := &dns.Client{Net: transport}
+	t := time.Duration(*recurserTimeout) * time.Second
+	c := &dns.Client{
+		Net:          transport,
+		DialTimeout:  t,
+		ReadTimeout:  t,
+		WriteTimeout: t,
```

Comment:
Should we use the combined `Timeout` instead?  It seems odd that I set a timeout of 2 seconds and it can take 2+2+2 seconds.